src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -112,13 +113,18 @@ class JesBackend extends Backend with LazyLogging {
   override def initializeForWorkflow(workflow: WorkflowDescriptor): HostInputs = workflow.actualInputs
 
   def taskOutputToJesOutput(taskOutput: TaskOutput, callGcsPath: String, scopedLookupFunction: ScopedLookupFunction, engineFunctions: JesEngineFunctions): Try[Option[JesOutput]] = {
-    taskOutput.wdlType match {
-      case WdlFileType =>
-        taskOutput.expression.evaluate(scopedLookupFunction, engineFunctions, interpolateStrings = true) match {
-          case Success(v) => Success(Option(JesOutput(taskOutput.name, s"$callGcsPath/${taskOutput.name}", Paths.get(v.valueString))))
-          case Failure(e) => Failure(new IllegalArgumentException(s"JES requires File outputs to be determined prior to running, but ${taskOutput.name} can not."))
-        }
-      case _ => Success(None)
+    // Currently, no function calls should be run before the JES call completes, so we can't retrieve files based on function calls.
+    if (taskOutput.expression.containsFunctionCall) {
+      Success(None)
+    } else {
+      taskOutput.wdlType match {
+        case WdlFileType =>
+          taskOutput.expression.evaluate(scopedLookupFunction, engineFunctions, interpolateStrings = true) match {
```

Comment:
I'll break my usual "all on one line" pattern and suggest that the `taskOutput.....` part get assigned to a val and then match on that